src/core/nixl_p2p_metadata_backend.cpp

```
@@ -0,0 +1,392 @@
+    int ret_fd = socket(AF_INET, SOCK_STREAM | SOCK_NONBLOCK, 0);
+    if (ret_fd == -1) {
+        NIXL_ERROR << "socket creation failed for ip_addr: " << ip_addr << " and port: " << port;
+        return -1;
+    }
+
+    int ret = connect(ret_fd, (struct sockaddr *)&listenerAddr, sizeof(listenerAddr));
+    if (ret < 0 && errno != EINPROGRESS) {
+        close(ret_fd);
+        return -1;
+    }
+
```

Cppcheck (2.21.0)

[style] The function 'lock' is never used.

(unusedFunction)